### PR TITLE
DR-916 Fix webapp tests

### PR DESCRIPTION
**Pull request by mvillaseco**
Changes:
 - Replace old Karma tests to Protractor.
 - Fix issue with tryErrorHandler.
 - Fix Language test.

### CODE REVIEW

File: src/spec/reports_spec.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        });
+        $httpBackend.whenGET(/\/accounts\/[\w|-]*\/statistics\/deliveries\/(by_hour|by_day)/).respond(200, {
+          "items": [ { "dropped": 3 }, { "dropped": 2 } ]
+        });
+      }));
+    var reportsPage = new ReportsPage();
+
+    // Act
+    browser.get('/#/reports');
+    expect(browser.getCurrentUrl()).toEqual(browser.baseUrl + '/#/reports');
```
Comment (nreal) [inline src/spec/reports_spec.js]: I think that this line should be under line 42 //Assert:

// Assert
expect(reportsPage.getDroppedEl()).toEqual('5');
expect(browser.getCurrentUrl()).toEqual(browser.baseUrl + '/#/reports');

File: src/spec/reports_spec.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        });
+        $httpBackend.whenGET(/\/accounts\/[\w|-]*\/statistics\/deliveries\/(by_hour|by_day)/).respond(200, {
+          "items": [ { "dropped": 0 }, { "dropped": 0 } ]
+        });
+      }));
+    var reportsPage = new ReportsPage();
+
+    // Act
+    browser.get('/#/reports');
+    expect(browser.getCurrentUrl()).toEqual(browser.baseUrl + '/#/reports');
```
Comment (nreal) [inline src/spec/reports_spec.js]: I think that this line should be under line 42 //Assert:

// Assert
expect(browser.getCurrentUrl()).toEqual(browser.baseUrl + '/#/reports');
expect(reportsPage.getDroppedEl()).toEqual('0');

**Comment by nreal** [comment]
It's Ok for me, good job

File: src/spec/reports_spec.js (hunk 1 above)
Comment (mvillaseco) [inline src/spec/reports_spec.js]: I'm currently using that line to set up the environment correctly in order to check for the value after, i'm not sure, if checking the url in the main assert of the test is correctly.

@andresmoschini  thoughts?

**Comment by andresmoschini** [comment]
Oh, I known that there was a a PR, but I did not see it in our old repo, sorry, I will review it is about an hour.

File: src/spec/onPrepare.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
+browser.driver.manage().window().setSize(1280, 960);
```
Comment (andresmoschini) [inline src/spec/onPrepare.js]: Could you add a `// TODO` mark to indicate that it is missing to set _saucelabs_ display size?

**Comment by andresmoschini** [comment]
:+1:

**Pull request merged**